snorkel/parser.py

```diff
@@ -288,22 +486,140 @@ def __init__(self, tok_whitespace=False, split_newline=False, parse_tree=False):
                         connect=20,
                         read=0,
                         backoff_factor=0.1,
-                        status_forcelist=[ 500, 502, 503, 504 ])
+                        status_forcelist=[500, 502, 503, 504])
         self.requests_session.mount('http://', HTTPAdapter(max_retries=retries))
-        
 
-    def _kill_pserver(self):
-        if self.server_pid is not None:
+    def _get_props(self, annotators, annotator_opts):
+        '''
+        Enable advanced configuration options for CoreNLP
+        Options are configured by each separate annotator
+
+        :param opts: options dictionary
+        :return:
+        '''
+        opts = []
+        for name in annotator_opts:
+            if not annotator_opts[name]:
+                continue
+            props = ["{}={}".format(key, str(value).lower()) for key, value in annotator_opts[name].items()]
+            opts.append('"{}.options":"{}"'.format(name, ",".join(props)))
+
+        props = []
+        props += ['"annotators": {}'.format('"{}"'.format(",".join(annotators)))]
+        props += ['"outputFormat": "json"']
+        props += [",".join(opts)] if opts else []
+        return ",".join(props)
+
+    def _ssplit_opts(self, split_newline=False, newline_sent_break="two"):
+        '''
+        Sentence splitter options
+
+        ssplit.eolonly
+        ssplit.isOneSentence
+        ssplit.newlineIsSentenceBreak  # "always", "never", or "two"
+        ssplit.boundaryMultiTokenRegex
+        ssplit.boundaryTokenRegex
+        ssplit.boundariesToDiscard
+        ssplit.htmlBoundariesToDiscard
+        ssplit.tokenPatternsToDiscard
+
+        :return:
+        '''
+        return {"newlineIsSentenceBreak": newline_sent_break}
+
+    def _ner_opts(self):
```

Comment:
Removed. Users will have to provide their own configuration dictionary to use any advanced options. Otherwise we just rely on CoreNLP defaults.